File: imagepicker/src/main/java/com/esafirm/imagepicker/helper/ImagePickerPreferences.java
package com.esafirm.imagepicker.helper;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class ImagePickerPreferences {

    public static final String PREF_WRITE_EXTERNAL_STORAGE_REQUESTED = "writeExternalRequested";
    public static final String PREF_CAMERA_REQUESTED = "cameraRequested";

    private Context context;

    public ImagePickerPreferences(Context context) {
        this.context = context;
    }

    /**
     * Set a permission is requested
     */
    public void setPermissionRequested(String permission) {
        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor editor = preferences.edit();
        editor.putBoolean(permission, true);
        editor.apply();
    }

    /**
     * Check if a permission is requestted or not (false by default)
     */
    public boolean isPermissionRequested(String permission) {
        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
        return preferences.getBoolean(permission, false);
    }
}

File: imagepicker/src/main/java/com/esafirm/imagepicker/helper/ImagePickerPreferences.kt
package com.esafirm.imagepicker.helper

import android.content.Context
import android.preference.PreferenceManager

class ImagePickerPreferences(context: Context) {

    companion object {
        private const val KEY_PERMISSION_GRANTED = "Key.WritePermissionGranted"
    }

    // TODO: Change this with data store
    private val preferences = PreferenceManager.getDefaultSharedPreferences(context)

    /**
     * Set a permission is requested
     */
    fun setPermissionIsRequested() {
        preferences.edit()
            .putBoolean(KEY_PERMISSION_GRANTED, true)
            .apply()
    }

    /**
     * Check if a permission is requestted or not (false by default)
     */
    fun isPermissionRequested(): Boolean {
        return preferences.getBoolean(KEY_PERMISSION_GRANTED, false)
    }
}